compiler/src/iree/compiler/Codegen/Dialect/GPU/Transforms/Transforms.cpp

```diff
@@ -113,10 +114,21 @@ static void replaceConsumerChain(RewriterBase &rewriter, Location loc,
                                  SmallVector<Operation *> consumerChain) {
   auto extractSlice = cast<tensor::ExtractSliceOp>(consumerChain.back());
   OpBuilder::InsertionGuard g(rewriter);
+  Value shuffleDest = parallelInsert.getDest();
+  if (auto empty = shuffleDest.getDefiningOp<tensor::EmptyOp>()) {
```

Comment:
We can just fail if this is not the case for now because if it doesn't end up with shared memory address space then it's a bug.